File: agilelink-view/src/src/app/cruds/project.service.js
(function() {
    'use strict';

    angular
        .module('crud-module')
        .factory('projectService', ProjectService);
        

    /* @ngInject */
    function ProjectService($http) {
    	var urlBase = '/rest/project' ;
        var service = {
        	archive: archive,
            save: save,
            load: load,
            remove: remove,
            find: find,
            list: list
        };
        return service;
        
        function save(project) {
        	return $http.post(urlBase+'/save', project);
        };
        
        function archive(project) {
        	return $http.post(urlBase+'/archive', project);
        };

        function load (id) {
        	return $http.put(urlBase+'/load/' + id);
        };

        function remove (id) {
            return $http.delete(urlBase + '/delete/' + id);
        };

        function list () {
            return $http.get(urlBase + '/list');
        };

        function find (searchObject) {
            return $http.post(urlBase + '/find', searchObject);
        };
    }
})();

File: agilelink-service/src/main/java/io/codelink/agilelink/service/ProjectService.java
package io.codelink.agilelink.service;

import java.util.Date;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.mongodb.core.MongoTemplate;
import org.springframework.stereotype.Service;

import io.codelink.agilelink.model.Project;
import io.codelink.agilelink.respository.ProjectRepository;
import io.codelink.agilelink.to.ProjectSearchTo;

@Service
public class ProjectService {

	@Autowired
	private ProjectRepository repository;
	
	@Autowired
	private MongoTemplate mongoTemplate;

	public Project load(String id) {
		return this.getRepository().findOne(id);
	}
	
	public Project save(Project project) {
		return this.getRepository().save(project);
	}
	
	public Page<Project> list(ProjectSearchTo searchTo) {
//		Query query = this.buildQuery(searchTo);
//		return this.getMongoTemplate().find(query, Project.class);
//		if(searchTo.isOnlyActives()) {
//			if(StringUtils.isEmpty(searchTo.getSearchTerm())) {
//				return this.getRepository().searchAllActive(searchTo);
//			}
//			return this.getRepository().searchAllActive(searchTo.getSearchTerm(), searchTo);
//		}
//		if(StringUtils.isEmpty(searchTo.getSearchTerm())) {
//			return this.getRepository().searchAll(searchTo);
//		}
//		return this.getRepository().searchAll(searchTo.getSearchTerm(), searchTo);
		return null;
	}

	public Iterable<Project> listAll() {
		return this.getRepository().findAll();
	}

	public Project archive(Project project) {
		if(project == null || project.getArchivationDate() != null) {
			return project;
		}
		project.setArchivationDate(new Date());
		return this.save(project);
	}
	
	protected ProjectRepository getRepository() {
		return repository;
	}

	protected MongoTemplate getMongoTemplate() {
		return mongoTemplate;
	}

	public Project activate(Project project) {
		if(project == null || project.getArchivationDate() == null) {
			return project;
		}
		project.setArchivationDate(null);
		return this.save(project);
	}
	
}